doc/specification.md

```diff
@@ -0,0 +1,597 @@
+
+#### 4.1.3.3 Life cycle of an AFT operation
+
+The life of an AFT operation starts when a client creates it, and ends in the
+following scenarios:
+
+*   The device failed to program the operation into RIB (returns `FAILED`).
+*   `ack_type` = `RIB_ACK`, the device programmed the operation into RIB
+    successfully (returns `RIB_PROGRAMMED`).
+*   `ack_type` = `RIB_AND_FIB_ACK`, the device programmed the operation into FIB
+    successfully (returns `FIB_PROGRAMMED`).
+*   `ack_type` = `RIB_AND_FIB_ACK`, the device has successfully programmed the
+    operation into RIB but failed to program it into FIB (returns `FIB_FAILED`).
+    Note that this is regardless if a device is going to retry the FIB
+    programming or not. The client can promptly send another AFT operation for
+    explicit behaviors (e.g. `ADD` for retry, and `DELETE` for stopping retry).
+*   The existing gRPC session is disconnected/canceled. All pending AFT
+    operations from the client should be cancelled.
+*   The device has discovered a change in the elected leader (see
+    [4.1.4.2](#4142-new-leader-election-in-single_primary) for more details).
+
+Only during the life cycle should the device keep the client updated via
+`AFTResult` message in `ModifyResponse`.
+
+### 4.1.4 Redundancy Mode
+
+`Modify` can operate in one of the following redundancy mode
+([4.1.1](#411-client-server-session-negotiation) defines how the mode is agreed
+between client and server):
+
+*   `SINGLE_PRIMARY`: The device accepts AFT operations from the primary client
+    only. The device discards AFT operations received from non-primary client
+    and respond error (see [4.1.2](#412-election-id) for details).
+*   `ALL_PRIMARY`: The device accepts AFT operations from all clients.
+
+#### 4.1.4.1 Client Election In `SINGLE_PRIMARY`
+
+gRIBI server does not participate in the election process, rather it consumes
+the election result. Election result is reflected in the
+`ModifyRequest.election_id` sent by clients. gRIBI server treats the client of
+the highest election ID as the primary client.
+
+When a client's election ID changed, the client should send a `ModifyRequest`
+with only the `ModifyRequest.election_id` populated. The device should respond
+with a `ModifyResponse` that has only the `election_id` field populated. The
+`ModifyResponse.election_id` by the server should be the highest election ID
+that the device has learnt from any client.
+
+If the `ModifyRequest.election_id` sent by a client matches the previous highest
+value, the newer client is considered primary. This allows for a client to
+reconnect without an external election having taken place. It is gRIBI clients'
+responsibility to avoid more than one `Modify` RPC sessions that are of the same
+highest Election ID, i.e., "dual-primary" situation in a `SINGLE_PRIMARY` mode.
+
+#### 4.1.4.2 New Leader Election In `SINGLE_PRIMARY`
+
+Switching to a new leader client occurs when a new client connects with
```

Comment:
Should we standardise on a single term for the leader/primary client? I think primary is fine (but I think we chose leader when we did an inclusive language review).

Reply:
Sounds great :-)